sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/implementation/models/ContainerRegistryRepositoriesGetTagsNextHeaders.java

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+
+package com.azure.containers.containerregistry.implementation.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/** The ContainerRegistryRepositoriesGetTagsNextHeaders model. */
+@Fluent
+public final class ContainerRegistryRepositoriesGetTagsNextHeaders {
+    /*
+     * The Link property.
+     */
+    @JsonProperty(value = "Link")
+    private String link;
```

Comment:
Why are there different returns types for the initial response and subsequent page responses? It appears to create twice the API surface (if these were public) for the same API.

Reply:
This is how our autogen generates code for the x-ms-pageable header. The APIs does not directly support pagination

---
In reply to: [599181026](https://github.com/Azure/azure-sdk-for-java/pull/20039#discussion_r599181026) [](ancestors = 599181026)